src/libcprover-rust/build.rs

```diff
@@ -111,28 +111,6 @@ fn main() {
         "cargo:rustc-link-search=native={}",
         libraries_path.display()
     );
-    println!("cargo:rustc-link-lib=static=goto-programs");
-    println!("cargo:rustc-link-lib=static=util");
-    println!("cargo:rustc-link-lib=static=langapi");
-    println!("cargo:rustc-link-lib=static=ansi-c");
-    println!("cargo:rustc-link-lib=static=analyses");
-    println!("cargo:rustc-link-lib=static=goto-instrument-lib");
-    println!("cargo:rustc-link-lib=static=big-int");
-    println!("cargo:rustc-link-lib=static=linking");
-    println!("cargo:rustc-link-lib=static=goto-checker");
-    println!("cargo:rustc-link-lib=static=solvers");
-    println!("cargo:rustc-link-lib=static=assembler");
-    println!("cargo:rustc-link-lib=static=xml");
-    println!("cargo:rustc-link-lib=static=json");
-    println!("cargo:rustc-link-lib=static=json-symtab-language");
-    println!("cargo:rustc-link-lib=static=cpp");
-    println!("cargo:rustc-link-lib=static=jsil");
-    println!("cargo:rustc-link-lib=static=statement-list");
-    println!("cargo:rustc-link-lib=static=goto-symex");
-    println!("cargo:rustc-link-lib=static=pointer-analysis");
-    for solver_lib in solver_libs {
-        println!("cargo:rustc-link-lib=static={}", solver_lib);
```

Comment:
(Okay, answering my own question here - it seems that we're building without issues, at least on the minisat front https://github.com/diffblue/cbmc/actions/runs/4293367669/jobs/7480953774)

Still I would be a bit apprehensive to remove this for loop. Can we at least comment it out so that we may re-introduce it if we find we need it for radical?

Reply:
The SAT solvers (dependency of the `solver` target) are added to the final API library.
Specifically they are gathered at line 25 of `src/libcprover-cpp/CMakeLists.txt` when doing `get_target_property(LIBRARY_DEPENDENCIES solvers LINK_LIBRARIES)`, and added afterwards.